nuxeo-connect-client/pom.xml

```diff
@@ -62,6 +62,11 @@
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.squareup.okhttp3</groupId>
+      <artifactId>mockwebserver</artifactId>
```

Comment:
I would rather not add an extra dependency and use the one already used in other test classes like https://github.com/nuxeo/nuxeo-connect/blob/master/nuxeo-connect-client/src/test/java/org/nuxeo/connect/connector/http/TestConnectHttpConnector.java#L56
I know it's simplier to use your usual lib but it's not good for maintenance ;)

Reply:
I tried at first to use what was in `TestConnectHttpConnector` but it appeared very complex to me and I didn't want to break anything inthere. What annoys me more precisely are:

 - some asserts are not made in the test, but in the custom Handler. When the assert breaks, the test ends up failing because the server returned a 500 error. Then the real error appears in the logs.
 - asserts happens in another thread... that's weird
 - I just wanted to add a `Set-Cookie` header: it would have required a new method in the custom handler class and a new `if` statement in the handler method: that may impact all existing tests, which breaks the unit test paradigm.
 - globally the logic of each test is not straightforward because some assertions happen in the test handler, so they have to be declared before the tested call. 
 - if the test breaks, I don't know if it's what I'm testing or the custom handler that breaks as it is not tested by itself.
 - as a result, the maintenance of that class is already hard for the one who did not code it, which defeats the purpose of unit tests

So instead of dealing with that, I thought that I could use and SHARE a very nice library which doesn't carry any real dependency (all deps are from the same project) and that is far easier to use, especially when using a Given / When / Then scheme. I naively thought that one could say: "oh this lib is nicer than what we have, probably we could use it from now on, and refactor when we have time"

Instead of just reporting a problem and waiting for a eventual fix, I created a test to reproduce it... and eventually also updated the faulty code. That kind of review just make me think that next time I'll stop after creating a ticket.

Now, as I'm a bit stubborn but also optimist, I did the refactor work, merged the two classes in another commit using only MockWebServer and removing the Jetty dependency. The result is far easier to read and understand than what was done before (also quicker to run). Do as you want, take it, trash it, on my side I wont spend any other minute on this.

Reply:
Hello damien, thank you for your answer AND for the refactoring of the old and badly coded test class. 
Let me just say that I never wanted to upset you and just wanted to share my first thought about adding a new dependency to the module (you tell me that you had the same, but I couldn't have know that). I didn't requested the change and approved the PR, which means I was confident you would have take my comment in account or not as you wished. I don't think this was rude or misplaced (if it was, please tell me how), so if you don't want to read this kind of reviews, just don't ask for it.